llarp/net/net.cpp

```diff
@@ -533,36 +537,45 @@ namespace llarp
     return std::nullopt;
   }
 
-  std::optional<llarp::Addr>
+  std::optional<IpAddress>
   GetIFAddr(const std::string& ifname, int af)
   {
     sockaddr_storage s;
     sockaddr* sptr = (sockaddr*)&s;
     if (!llarp_getifaddr(ifname.c_str(), af, sptr))
       return std::nullopt;
-    return llarp::Addr{*sptr};
+    // TODO: why the fuck does this not compile?
+    // llarp::SockAddr saddr = SockAddr(*sptr);
```

Comment:
FYI this might be a compiler bug. I get errors about missing definition for constructor that takes `sockaddr&`, even when the definition was provided. I couldn't reproduce in a test case. If I provided the definition with the declaration (e.g. in the `.hpp` file) the compiler was happy.